helpers/setupjsongenerator.py

```diff
@@ -0,0 +1,675 @@
+        def remove_storagerouter
+        def add_disk_to_sr
+        def remove_disk_from_sr
+        def add_domain_to_sr
+        def remove_domain_from_sr
+
+        def add_backend
+        def remove_backend
+        def add_preset_to_backend
+        def remove_preset_to_backend
+        def add_osd_to_backend
+
+        def add_vpool
+        def remove_vpool
+
+        def change_cache
+
+    """
+    HYPERV_KVM = 'KVM'
+    VPOOL_COUNTER = 1
+
+    def __init__(self):
+        self._json_dict = {}
+        self._presets = []
+        self._domains = []
+        self._backends = []
+        self._ips = []
+
+    @property
+    def config(self):
+        """
+        Property containing the currently modelled config dict
+        :return: return the currently modelled config dict
+        :rtype: dict
+        """
+        return self._json_dict
+
+    def dump_json_to_file(self, path):
+        """
+        Write current setup dict to a json file in the provided path.
+        :param path: path to dump json file to
+        :type path: str
+        """
+        with open(path, 'w') as fp:
+            json.dump(self.config, indent=4, sort_keys=True, fp=fp)
+
+    def update_scenarios(self, scenarios=None):
+        """
+        Add scenarios to be scheduled in the setup.
+        :param scenarios:
+        :type scenarios: list
+        """
+        if not isinstance(scenarios, list) and scenarios is not None:
+            raise ValueError('Scenarios should be passed in a list format, not {}'.format(type(scenarios)))
+
+        if scenarios is None:
+            self.config['scenarios'] = ['ALL']
+        else:
+            for scenario in scenarios:
+                if isinstance(scenario, str) and scenario not in AutoTests.list_tests():
+                    raise ValueError('Scenario {0} is not a valid scenario path.'.format(scenario))
+
+            self.config['scenarios'] = scenarios
+
+    def update_ci(self, passed_required_params, passed_optional_params=None):
+        """
+        Set the ci constants of the setup file accordign to the passed parameters.
+        :param passed_required_params: obligatory parameters for the setup file
+        :type passed_required_params: dict
+        :param passed_optional_params: optional parameters
+        :type passed_optional_params: dict
+
+        """
+        required_params_layout = {'setup': (bool, None, True),
+                                  'grid_ip': (str, Toolbox.regex_ip, True)}
+
+        default_params = {'validation': False,
+                          'cleanup': False,
+                          'send_to_testrail': True,
+                          'fail_on_failed_scenario': True,
+                          'scenarios': True,
+                          'version': 'andes',
+                          'config_manager': 'arakoon'}
+
+        if passed_optional_params is None:
+            passed_optional_params = {}
+
+        for key, value in default_params.iteritems():
```

Comment:
If specifying your own necessary/optional params, I'd expect being able to overrule the default ones too
use dict.update instead (this will override the keys for which a value already exists)